dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/FileManageE2ETest.java

```diff
@@ -116,4 +178,112 @@ void testDeleteDirectory() {
+            .as("File list should contain newly-created file")
+            .extracting(WebElement::getText)
+            .anyMatch(it -> it.contains(testFileName)));
+    }
+
+    @Test
+    @Order(41)
+    void testRenameFile() {
+        final FileManagePage page = new FileManagePage(browser);
+
+        page.rename(testFileName, testRenameFileName);
+
+        await().untilAsserted(() -> {
+            browser.navigate().refresh();
+
+            assertThat(page.fileList())
+                .as("File list should contain newly-created file")
+                .extracting(WebElement::getText)
+                .anyMatch(it -> it.contains(testRenameFileName));
+        });
+    }
+
+    @Test
+    @Order(42)
+    void testEditFile() {
+        final FileManagePage page = new FileManagePage(browser);
+        String scripts = "echo 456";
+
+        page.editFile(testRenameFileName, scripts);
+
+        await().untilAsserted(() -> assertThat(page.fileList())
+            .as("File list should contain newly-created file")
+            .extracting(WebElement::getText)
+            .anyMatch(it -> it.contains(testRenameFileName)));
+    }
+
+    @Test
+    @Order(45)
+    void testDeleteFile() {
+        final FileManagePage page = new FileManagePage(browser);
+
+        page.delete(testRenameFileName);
+
+        await().untilAsserted(() -> {
+            browser.navigate().refresh();
+
+            assertThat(
+                page.fileList()
+            ).noneMatch(
+                it -> it.getText().contains(testRenameFileName)
+            );
+        });
+    }
+
+    @Test
+    @Order(60)
+    void testUploadOver1GBFile() throws IOException {
+        final FileManagePage page = new FileManagePage(browser);
+
+        RandomAccessFile file = new RandomAccessFile(testOver1GBFilePath, "rw");
+        file.setLength((long) (1.5 * 1024 * 1024 * 1024));
+
+        page.uploadFile(testOver1GBFilePath);
+
+        await().untilAsserted(() ->
+            assertThat(browser.findElement(By.tagName("body")).getText())
+                .contains("Upload File size cannot exceed 1g")
+        );
+    }
+
+    @Test
+    @Order(65)
+    void testUploadUnder1GBFile() throws IOException {
+        final FileManagePage page = new FileManagePage(browser);
+
+        browser.navigate().refresh();
+
+        RandomAccessFile file = new RandomAccessFile(testUnder1GBFilePath, "rw");
+        file.setLength((long) (0.01 * 1024 * 1024 * 1024));
+
+        page.uploadFile(testUnder1GBFilePath);
+
+        await().untilAsserted(() -> {
+            assertThat(page.fileList())
+                .as("File list should contain newly-created file")
+                .extracting(WebElement::getText)
+                .anyMatch(it -> it.contains(testUnder1GBFileName));
+        });
+    }
+
+    @Test
+    @Order(70)
+    void testDownloadFile() {
+        final FileManagePage page = new FileManagePage(browser);
+
+        page.downloadFile(testUnder1GBFileName);
+
```

Comment:
Can you add an assertion to assert that the file exists locally after the download operation?

Reply:
The file is downloaded in selenium/standalone-chrome-debug container. But the E2E test code runs on the host. This makes it difficult to monitor the existence of downloaded files. Is it a good way to mount ```/home/runner/download``` to the host? Or Is there a better way?

Reply:
This is not the only case where we need to download files in E2E, I think we will have more in the future so we need a solution.

What do you think if we

1. Change the default download directory of the browser to a static one if it's not already, like what you said `/home/runner/download`.
2. Mount the directory by modifying https://github.com/apache/dolphinscheduler/blob/5be8e0531c4ab4863f0ec5d590418014575b0bfc/dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java#L119-L121 like this

```java
        browser = new BrowserWebDriverContainer<>()
            .withCapabilities(new ChromeOptions())
            .withFileSystemBind(Paths.get(System.getProperty("java.io.tmpdir"), "download").toFile().getAbsolutePath(), "/home/download")
            .withRecordingMode(RECORD_ALL, record.toFile(), MP4);
```

So that we can find the downloaded files in `Paths.get(System.getProperty("java.io.tmpdir"), "download").toFile().getAbsolutePath()` and assert what we want.

> You can encapsulate the `Paths.get(System.getProperty("java.io.tmpdir"), "download").toFile().getAbsolutePath()` to a more convenient method like a utility class or similar if you want